bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -267,6 +273,49 @@ public void getURLInput(String url) throws IOException {
         }
     }
 
+    /**
+     * Builds the {@link TimeSeries} that represents the future tibber prices.
+     * As they are valid for one hour every entry is represented by two time series entries.
+     * 
+     * @param today The prices for today
+     * @param tomorrow The prices for tomorrow.
+     * @return The {@link TimeSeries} with future values.
+     */
+    private TimeSeries buildTimeSeries(JsonArray today, JsonArray tomorrow) {
+        final TimeSeries timeSeries = new TimeSeries(TimeSeries.Policy.REPLACE);
+
+        for (JsonElement entry : today) {
+            DateTimeType startsAt = parseDateTime(entry.getAsJsonObject().get("startsAt"));
+            timeSeries.add( //
+                    startsAt.getInstant(), //
+                    new DecimalType(entry.getAsJsonObject().get("total").getAsString()) //
+            );
+            timeSeries.add( //
+                    startsAt.getInstant().plus(1, ChronoUnit.HOURS).minus(1, ChronoUnit.MICROS), //
+                    new DecimalType(entry.getAsJsonObject().get("total").getAsString()) //
+            );
```

Comment:
This seems a little odd, why are you adding the same entry again with a new timestamp?

Reply:
Because this is the semantics of the api: The given price starts at the given time and is valid for exactly one hour (when the next price starts). If the second entry is not present, openhab assumes (and displays) a linear curve, which is wrong. This way it's display just the way i want and how its displayed whitin the tibber app: as stepping function:
<img width="957" alt="Tibber preview" src="https://github.com/openhab/openhab-addons/assets/24605357/79f76d00-a177-4db8-8ea9-d6969055ea06">
I've added a comment in source code, too.

Reply:
This is how such line charts are generated. You can create a bar chart instead for a better representation. Without time series you would have the same issue when the current price is updated once per hour and looking at a chart back in time. I understand what you are trying to accomplish, but I don't think it's the right way. Logically, this work-around causes these events:
- 14.01.2023 12:00:00.000 1 EUR
- 14.01.2023 12:59:59.999 1 EUR
- 14.01.2023 13:00:00.000 1,1 EUR
- 14.01.2023 13:59:59.999 1,1 EUR

It shouldn't be needed to push an item state change to the event bus when the value didn't change. So I think this problem is in how you created the chart, and it would be the same issue for all line charts. Maybe there are some options for modifing line charts in this regard, but I'm not strong in this area.

Reply:
@J-N-K - do you have an opinion on this matter?

Reply:
I've found an option for the apache eChart to display a step line:
https://echarts.apache.org/en/option.html#series-line.step
So this fixes the issue a bette way.
Thanks for your input!